b2dEditor.py

```diff
@@ -0,0 +1,347 @@
+                            set_current_shape(shapes.index(nodes))
+
+                if origin.pos[0] - 6 <= event.pos[0] <= origin.pos[0] + 6 and origin.pos[1] - 6 <= event.pos[1] <= origin.pos[1] + 6:
+                    selected = origin
+
+                if selected == None:
+                    if edit_area.rect.collidepoint(mouse_pos):
+                        add_node(event.pos)
+
+            # Check if mouse2 (3 in pygame terms) clicked on node
+            elif event.button == 3:
+                for nodes in shapes:
+                    for i in nodes:
+                        # Compare event position to each node's position
+                        if i.pos[0] - 5 <= event.pos[0] <= i.pos[0] + 5 and i.pos[1] - 5 <= event.pos[1] <= i.pos[1] + 5:
+                        # Remove clicked node
+                            nodes.remove(i)
+
+        # Set selected node's position to the mouse's
+        elif event.type == pygame.MOUSEMOTION:
+            if selected != None:
+                if edit_area.rect.collidepoint(mouse_pos):
+                    selected.set_pos(event.pos)
+                else: selected = None
+
+
+        # De-select node on mouse1 release
+        elif event.type == pygame.MOUSEBUTTONUP:
+            if event.button == 1:
+                if selected != None:
+                    selected.clicked = False
+                selected = None
+
+
+def set_current_shape(num):
+    global current_shape
+    current_shape = num
+
+def add_node(pos):
+    new_node = Node()
+    new_node.set_pos(pos)
+    shapes[current_shape].append(new_node)
+
+def add_shape():
+    shapes.append([])
+    set_current_shape(len(shapes) - 1)
+
+
+def run():
+    global edit_area, origin, mouse_pos, quit
+    pygame.init()
+    pygame.font.init()
+    font = pygame.font.Font(pygame.font.match_font("monospace,ubuntumono,liberationmono"), int(resolution[1] / 40))
+    disp = pygame.display.set_mode(resolution)
+    screen = pygame.Surface(resolution)
+    edit_area = NodeRect()
+    ticker = pygame.time.Clock()
+    origin = OriginNode()
+    origin.set_pos((100, 100))
+    button_x = resolution[0] / 4
+    button_y = resolution[1] / 16
+    Button((button_x, button_y), (resolution[0] - button_x, button_y), "set_background()", "Set Background Image")
+    Button((button_x, button_y), (resolution[0] - button_x, button_y * 2.1), "dump_node_json()", "Save JSON")
+    Button((button_x, button_y), (resolution[0] - button_x, button_y * 3.2), "load_node_json()", "Load JSON")
+    Button((button_x, button_y), (resolution[0] - button_x, button_y * 4.3), "add_shape()", "Add Object")
+    Button((button_x, button_y), (resolution[0] - button_x, resolution[1] - button_y * 2), "exit()", "Exit")
+    while not quit:
+
+        ticker.tick(24)
+        # Clear the screen
+        screen.fill((200, 200, 200))
+
+        mouse_pos = pygame.mouse.get_pos()
+
+        # Handle inputs
+        for event in pygame.event.get():
+
```

Comment:
Likewise could see this split off into a separate method

Reply:
Consider splitting this into a separate method, named, for instance, `handle_inputs()`